### PR TITLE
Add wheel zoom and copy / paste to core-example-advanced

### PR DESCRIPTION
**Pull request by brianfoody**
Integrating with existing events to enable wheel zoom.

And a COPY and PASTE event and handlers to do copy and paste.

### CODE REVIEW

File: examples/core-example-advanced/src/App.tsx [hunk 1]
```diff
@@ -121,11 +129,24 @@ const onKeyDown: TLKeyboardEventHandler = (key, info, e) => {
       }
       break
     }
-    case 's':
-    case 'v': {
```
Comment (brianfoody) [inline examples/core-example-advanced/src/App.tsx:125]: v is a separate handler now to handle the case of CTRL+V

**Comment by judicaelandria** [comment]
@brianfoody error from vercel
Screen Shot 2022-09-20 at 10 40 04 AM

**Comment by brianfoody** [comment]
@judicaelandria thanks! Lost some things in my git goofing around. Have addressed now.

**Comment by judicaelandria** [comment]
Great work @brianfoody , 
NIT:just one thing can you please move the duplicate element a lil bit like the point: [shape.point + 50, shape.point + 50]

File: examples/core-example-advanced/src/state/actions/selection/paste.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { nanoid } from 'nanoid'
+import type { Action } from 'state/constants'
+
+export const paste: Action = (data) => {
+  const selectedShapes = Object.values(data.page.shapes).filter((shape) => shape.isCopied)
+
+  const newShapes = selectedShapes.map((s) => ({ ...s, id: nanoid() }))
```
Comment (judicaelandria) [inline examples/core-example-advanced/src/state/actions/selection/paste.ts:7]: you can do it like this
```ts
const newShapes = selectedShapes.map((s) => ({
    ...s,
    id: nanoid(),
    point: [s.point[0] + 10, s.point[1] + 10],
  }))```

Comment (brianfoody) [reply examples/core-example-advanced/src/state/actions/selection/paste.ts]: That's a lovely addition @judicaelandria , added now.

**Review by judicaelandria — APPROVED** [review]

**Comment by steveruizok** [comment]
Hey, thanks for the PR!

There are some problems with both features, so I'm going to hold off on merging.

For zooming: If you want to use scroll wheel for zooming, you'll need to smooth out the differences between how different browsers or devices apply zoom. See tldraw's scrolling for an example of how to normalize that delta. Sending ZOOM_IN / ZOOM_OUT events isn't really enough.

For copy and paste: if you waiting until you paste the shapes to create the copies, you run the risk of the copied shape having been since deleted. I'm guessing that a CUT event is not implemented here for that reason. Better to dump the shapes into memory somewhere and pull them back on paste. Even better to serialize them and store them in memory (e.g. localStorage) to allow for cross-tab pasting; and even better yet to attach them to the copy event to allow cross-browser pasting. Of course, I think in this case just putting them in memory would be better!

**Comment by brianfoody** [comment]
Thanks great feedback, I'll work on that and get back to you with the fixes.

**Pull request closed without merging**